packages/jest-core/src/testSchedulerHelper.ts

```diff
@@ -27,25 +27,25 @@ export function shouldRunInBand(
   }
 
   /*
-   * Run in band if we only have one test or one worker available, unless we
-   * are using the watch mode, in which case the TTY has to be responsive and
-   * we cannot schedule anything in the main thread. Same logic applies to
-   * watchAll.
+   * If we are using watch/watchAll mode, don't schedule anything in the main
+   * thread to keep the TTY responsive and to keep the watcher running even if
+   * the test crashes.
+   */
+  const isWatchMode = watch || watchAll;
+  if (isWatchMode) {
+    return false;
+  }
+
+  /*
+   * Otherwise, run in band if we only have one test or one worker available.
    * Also, if we are confident from previous runs that the tests will finish
    * quickly we also run in band to reduce the overhead of spawning workers.
-   * Finally, the user can provide the runInBand argument in the CLI to
-   * force running in band.
    * https://github.com/facebook/jest/blob/700e0dadb85f5dc8ff5dac6c7e98956690049734/packages/jest-config/src/getMaxWorkers.js#L14-L17
```

Comment:
Leave this comment in place, please. Check that link, the logic there sets `maxWorkers` to 1 if `runInBand` is provided. At first it is not clear why `runInBand` is mentioned here, but it makes sense. Or?

Reply:
Ah, you're right, thanks. (It seems with the `workerIdleMemoryLimit` check that is no longer quite right -- we should pass `runInBand` in here or more likely reject setting the two together -- but that's probably best left to another PR.)

Reply:
`workerIdleMemoryLimit` is a special case. It works only if workers are involved. There is no other mechanism to force workers. By the way, `workerIdleMemoryLimit` could be used to force workers in watch mode ;D